core/src/main/java/com/graphhopper/GraphHopper.java

```diff
@@ -858,39 +861,45 @@ public boolean load() {
 
         if (!allowWrites && dataAccessDefaultType.isMMap())
             dataAccessDefaultType = DAType.MMAP_RO;
-        if (encodingManager == null)
-            // we did not call init(), so we build the encoding manager now.
-            // just like when calling init, users have to make sure they use the same setup for import and load
-            buildEncodingManagerAndOSMParsers(flagEncodersString, encodedValuesString, dateRangeParserString, profilesByName.values());
 
         GHDirectory directory = new GHDirectory(ghLocation, dataAccessDefaultType);
         directory.configure(dataAccessConfig);
-        baseGraph = new BaseGraph.Builder(getEncodingManager())
-                .setDir(directory)
-                .set3D(hasElevation())
-                .withTurnCosts(encodingManager.needsTurnCostsSupport())
-                .setSegmentSize(defaultSegmentSize)
-                .build();
         properties = new StorableProperties(directory);
-        checkProfilesConsistency();
 
-        if (!new File(ghLocation).exists())
+        if (!new File(ghLocation).exists()) {
+            // there is nothing to load, so we create the EM from the config
+            buildEncodingManagerAndOSMParsers(flagEncodersString, encodedValuesString, dateRangeParserString, profilesByName.values());
+            baseGraph = new BaseGraph.Builder(getEncodingManager())
+                    .setDir(directory)
+                    .set3D(hasElevation())
+                    .withTurnCosts(encodingManager.needsTurnCostsSupport())
+                    .setSegmentSize(defaultSegmentSize)
+                    .build();
+            checkProfilesConsistency();
             return false;
+        }
 
         GHLock lock = null;
         try {
             // create locks only if writes are allowed, if they are not allowed a lock cannot be created
             // (e.g. on a read only filesystem locks would fail)
-            if (baseGraph.getDirectory().getDefaultType().isStoring() && isAllowWrites()) {
+            if (directory.getDefaultType().isStoring() && isAllowWrites()) {
                 lockFactory.setLockDir(new File(ghLocation));
                 lock = lockFactory.create(fileLockName, false);
                 if (!lock.tryLock())
                     throw new RuntimeException("To avoid reading partial data we need to obtain the read lock but it failed. In " + ghLocation, lock.getObtainFailedReason());
             }
-
-            if (!loadBaseGraphAndProperties())
+            if (!properties.loadExisting())
                 return false;
```

Comment:
I just realized that this could be problematic: When there is a -gh folder, but it is empty (no properties file) we will return here but no encoding manager and no base graph was created. Can we move the `properties.loadExisting()` call further up, before obtaining the read lock? Or can we create the lock earlier? And do we really need this lock?

Reply:
> What would be the problem in that case?

Right now we get an `IllegalStateException` when the -gh folder exists, but the properties file is missing or the entire folder is empty. The reason is that `baseGraph` is null and we check for null in the `process` method, so actually it is rather like a `NullPointerException`. 

> We have no other protection against multiple writes into the same folder.

Ok, and that protects us in case there are multiple GH processes using the same folder or why do we need this protection? But here we also do not need to write anything, we just want to read the contents of the folder and properties file (if they exist). Would it be OK to call `properties.loadExisting()` before obtaining the lock? I still think there should simply be static methods that tell us whether a certain file exists or allow us to load it without creating any DataAccess objects (let alone without obtaining any (write?) locks :), but this is for another issue...).

Reply:
I fixed the issue here: cadbca1c39a4a11c8c4fa9dd87d4ad3e078e24fc, so no when we import into an empty (but existing) folder everything works. Still, the code is quite twisted :)